From c12b5509333b4e1ac1fd0029cfa7e5e342a6931b Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hmehrtens@maxlinear.com>
Date: Tue, 12 Oct 2021 18:07:24 +0200
Subject: bridge: Change request for all bridge attributes

We have to change the change request if any of the attributes on the
bridge port is changed not only the flags.
Without this change the kernel will ignore the netlink command if only
cost, priority or port state is changed.

Fixes: 3c427d0f4fcc ("Change rtnl_link_af_ops.ao_override_rtm behavior")
Signed-off-by: Hauke Mehrtens <hmehrtens@maxlinear.com>
---
 lib/route/link/bridge.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/lib/route/link/bridge.c b/lib/route/link/bridge.c
index c5231a1..e0bdbd2 100644
--- a/lib/route/link/bridge.c
+++ b/lib/route/link/bridge.c
@@ -343,7 +343,10 @@ static int bridge_override_rtm(struct rtnl_link *link) {
 
         bd = bridge_data(link);
 
-        if (bd->ce_mask & BRIDGE_ATTR_FLAGS)
+        if (bd->ce_mask & BRIDGE_ATTR_FLAGS ||
+            bd->ce_mask & BRIDGE_ATTR_COST ||
+            bd->ce_mask & BRIDGE_ATTR_PRIORITY ||
+            bd->ce_mask & BRIDGE_ATTR_PORT_STATE)
                 return 1;
 
         return 0;
